src/TetherComponent.jsx

```diff
@@ -96,6 +96,15 @@ class TetherComponent extends Component {
       this._createContainer();
     }
 
+    // Verify if className props have changed
+    if (prevProps.className !== this.props.className) {
+      const prevClasses = (prevProps.className || '').split(' ');
```

Comment:
This will need some tests but here's the general idea of what probably needs to happen to fix the re-render issue